src/main/java/no/fint/portal/customer/controller/AssetController.java

```diff
@@ -63,6 +63,7 @@ public AssetController(PortalApiService portalApiService, AssetService assetServ
         // TODO: 31/07/2018 This should be moved to the portal-api
         asset.setAssetId(String.format("%s.%s", asset.getAssetId(), primaryAsset.getAssetId()));
 
+        if (isIllegalAssetID(asset.getAssetId())) throw new IllegalArgumentException("The assetId contains illegal characters: " + asset.getAssetId());
```

Comment:
Ta en titt på https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html

For eksempel kan `ContainsOnly` eller `IsAsciiPrintable`. brukes.

Reply:
ContainsOnly kan brukes, men da må jeg skrive alle godkjente tegn. Eksempel: `StringUtils.ContainsOnly("abcdefghijklmnopqrstuvwxyzABC...."` 
 
IsAsciiPrintable sjekker chars fra 32 til 126, i så fall må jeg gjøre en ekstra sjekk på symboler:
    private boolean isIllegalAssetID(String assetId) {
```
        if (StringUtils.isBlank(assetId)) return true;
        if (!StringUtils.isAsciiPrintable(assetId)) return true;
        if (StringUtils.containsAny(assetId, " !\"#$%&'()*+,/:;<=>?@[\\]^`{}|~")) return true;
        return false;
```
Hva syntes du er best @asgeirn? 
Hva med norske tegn, æøå, bør det være lov?

Reply:
Norske tegn bør ikke være lov nei.

Du kan også forkorte uttrykket og ta bort `if` ved å sette det sammen sånn:

```java
return StringUtils.isBlank(assetId)
    || !StringUtils.isAsciiPrintable(assetId)
    || StringUtils.containsAny(assetId, " !\"#$%&'()*+,/:;<=>?@[\\]^`{}|~");
```

Reply:
Bedre nå?